pkg/middleware/middleware.go

```diff
@@ -0,0 +1,153 @@
+package middleware
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/bketelsen/buffet"
+	"github.com/gobuffalo/buffalo"
+	"github.com/gomods/athens/pkg/config/env"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/log"
+	"github.com/gomods/athens/pkg/module"
+	"github.com/gomods/athens/pkg/paths"
+	"github.com/sirupsen/logrus"
+)
+
+type middlewareFunc func(entry log.Entry) buffalo.MiddlewareFunc
+
+// LogEntryMiddleware builds a log.Entry applying the request parameter to the given
+// log.Logger and propagates it to the given MiddlewareFunc
+func LogEntryMiddleware(middleware middlewareFunc, lggr *log.Logger) buffalo.MiddlewareFunc {
```

Comment:
can you separate those so we have 
- middleware/filter.go
- middleware/log_entry.go
- middleware/validation.go
- middleware/middleware_test.go